cartography/intel/trivy/scanner.py

```diff
@@ -361,3 +374,61 @@ def sync_single_image_from_s3(
             f"Failed to process S3 scan results for {image_uri} from {s3_object_key}: {e}"
         )
         raise
+
+
+@timeit
+def read_scan_results_from_file(
+    file_path: str,
+) -> tuple[str | None, list[dict], str | None]:
+    """Read and parse Trivy scan results from a local file."""
+    logger.debug(f"Reading scan results from file: {file_path}")
+    with open(file_path, encoding="utf-8") as f:
+        trivy_data = json.load(f)
+
+    artifact_name = trivy_data.get("ArtifactName")
+
+    if "Results" in trivy_data and trivy_data["Results"]:
+        results = trivy_data["Results"]
+    else:
+        stat_handler.incr("image_scan_no_results_count")
+        logger.warning(
+            f"File scan data did not contain a `Results` key for {file_path}; continuing."
+        )
+        results = []
```

Comment:
When Trivy has 0 findings, IIRC the expected behavior is for the `Results` key to exist but have an empty list. If the `Results` key is not present at all, I think that's an anomaly where we should raise an exc to stop the graph from continuing with the cleanup jobs.

Reply:
done